sdcm/cluster.py

```diff
@@ -1080,3 +950,523 @@ def verify_stress_thread(self, queue, db_cluster):
             self._cassandra_stress_plot(lines, plotfile, node, db_cluster)
 
         return errors
+
+
+class LibvirtCluster(BaseCluster):
+
+    """
+    Cluster of Node objects, started on Libvirt.
+    """
+
+    def __init__(self, domain_info, hypervisor, cluster_uuid=None,
+                 cluster_prefix='cluster',
+                 node_prefix='node', n_nodes=10, params=None):
+        self._domain_info = domain_info
+        self._hypervisor = hypervisor
+        super(LibvirtCluster, self).__init__(cluster_uuid=cluster_uuid,
+                                             cluster_prefix=cluster_prefix,
+                                             node_prefix=node_prefix,
+                                             n_nodes=n_nodes,
+                                             params=params)
+
+    def __str__(self):
+        return 'LibvirtCluster %s (Image: %s)' % (self.name,
+                                                  os.path.basename(self._domain_info['image']))
+
+    def add_nodes(self, count, user_data=None):
+        del user_data
+        nodes = []
+        os_type = self._domain_info['os_type']
+        os_variant = self._domain_info['os_variant']
+        memory = self._domain_info['memory']
+        bridge = self._domain_info['bridge']
+        uri = self._domain_info['uri']
+        image_parent_dir = os.path.dirname(self._domain_info['image'])
+        for index in range(count):
+            index += 1
+            name = '%s-%s' % (self.node_prefix, index)
+            dst_image_basename = '%s.qcow2' % name
+            dst_image_path = os.path.join(image_parent_dir, dst_image_basename)
+            self.log.info('Copying %s -> %s',
+                          self._domain_info['image'], dst_image_path)
+            shutil.copyfile(self._domain_info['image'], dst_image_path)
+            virt_install_cmd = ('virt-install --connect %s --name %s '
```

Comment:
As we talked in gtalk, I should setup libvirtd to disable auth.
I changed following parameters, then the patches works.

    auth_unix_ro = "none"
    auth_unix_rw = "none"
    auth_tcp = "none"
    auth_tls = "none"
    access_drivers = [ "none" ]


Reviewed-by: Amos Kong <amos@scylladb.com>
Tested-by: Amos Kong <amos@scylladb.com>